Python != CPython

njsmith opened the pull request:
[this probably doesn't work yet, but I was getting weird "network is unreachable" errors trying to build the image locally -- remember, the great thing about docker build is that it produces identical results no matter where you run it! -- so pushing up to see where travis breaks]

As we are reminded by #38, CPython is not the only implementation of
Python around. This PR doesn't add support for PyPy or anything else,
but it does rearrange things a bit to make room for them.

User visible changes:
- the full-version-number CPython build directories are moved into
  /opt/_internal to make clear that users shouldn't rely on the details
  of their names. Also, they're called e.g. cpython-2.6.9 now instead of
  just 2.6.9.
- The entries in /opt/python/ are now exactly the ABI tags used by each
  version. So we have e.g. /opt/python/cp27mu which generates wheels
  named like *-py27-cp27mu-manylinux1_*.whl. If you want to build
  your wheel against all CPythons, that's /opt/python/cp*/bin/python.

njsmith commented:
huh, it worked

ogrisel commented:
LGTM.

ogrisel commented:
The README files needs to be updated to take that new folder structure into account.

njsmith commented:
Updated README, but also changed my mind about what to call the directories :-) See the last commit message for details, and we'll see whether or not the build passes...

ogrisel commented:
Alright, LGTM too :)

njsmith commented:
@ogrisel: want to merge, then? You should have the power :-)

ogrisel commented:
Done!
